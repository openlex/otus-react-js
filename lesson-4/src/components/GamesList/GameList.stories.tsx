import * as React from "react";
import { gamesMock as games } from "@mocks";
import { GamesList } from "@components";

export default {
  title: "Список игр",
  component: GamesList,
};

export const List = () => <GamesList games={games} />;
