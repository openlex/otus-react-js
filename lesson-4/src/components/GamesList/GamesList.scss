.block {
}

.game {
	margin-bottom: 20px;
}
