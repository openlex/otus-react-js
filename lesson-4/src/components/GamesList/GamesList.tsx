import React from "react";
import * as s from "./GamesList.scss";
import { Game } from "@components";
import { IGame } from "@types";

interface IGamesListProps {
  games: IGame[];
}

export class GamesList extends React.Component<IGamesListProps> {
  render() {
    const { games } = this.props;

    return (
      <div className={s.block}>
        {games.map((game: IGame) => (
          <Game
            className={s.game}
            key={game.id}
            id={game.id}
            title={game.title}
            description={game.description}
            genres={game.genres}
            rating={game.rating}
          />
        ))}
      </div>
    );
  }
}
