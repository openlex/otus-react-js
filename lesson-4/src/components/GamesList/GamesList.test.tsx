import React from "react";
import { GamesList } from "./GamesList";
import { shallow } from "enzyme";
import { gamesMock as games } from "@mocks";

describe("GamesList", () => {
  const genreList = shallow(<GamesList games={games} />);

  it("Check contained games number", () => {
    expect(genreList.find("Game").length).toBe(2);
  });

  it("Snap shot", () => {
    expect(genreList).toMatchSnapshot();
  });
});
