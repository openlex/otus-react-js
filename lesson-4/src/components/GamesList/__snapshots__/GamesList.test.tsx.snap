// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesList Snap shot 1`] = `ShallowWrapper {}`;
